src/lib.rs

```diff
@@ -0,0 +1,201 @@
+                let log = format!($($arg)*);
+                let _ = log_to_redis(&log).await;
+            }
+        }
+    };
+
+}
+
+#[macro_export]
+macro_rules! warn {
+    ($location:expr,$($arg:tt)*) => {
+        match $location {
+            LogLocation::Local => {
+                tracing::warn!($($arg)*);
+            }
+            LogLocation::Redis => {
+                let log = format!($($arg)*);
+                let _ = log_to_redis(&log).await;
+            }
+            LogLocation::Both => {
+                tracing::warn!($($arg)*);
+                let log = format!($($arg)*);
+                let _ = log_to_redis(&log).await;
+            }
+        }
+    };
+
+}
+
+#[macro_export]
+macro_rules! error {
+    ($location:expr,$($arg:tt)*) => {
+        match $location {
+            LogLocation::Local => {
+                tracing::error!($($arg)*);
+            }
+            LogLocation::Redis => {
+                let log = format!($($arg)*);
+                let _ = log_to_redis(&log).await;
+            }
+            LogLocation::Both => {
+                tracing::error!($($arg)*);
+                let log = format!($($arg)*);
+                let _ = log_to_redis(&log).await;
+            }
+        }
+    };
+}
+
+pub enum LogLocation {
+    Local,
+    Redis,
+    Both,
+}
+
+lazy_static! {
+    static ref CLIENT_ID: Arc<RwLock<String>> = Arc::new(RwLock::new(String::new()));
+    static ref REDIS_POOL: Arc<RwLock<Option<Pool<RedisConnectionManager>>>> =
+        Arc::new(RwLock::new(None));
+}
+
+pub async fn init_redis_connection(
+    ip_addr: IpAddr,
+    port: u16,
+    client_id: String,
+) -> anyhow::Result<()> {
+    let manager = RedisConnectionManager::new(format!("redis://{}:{}", ip_addr, port))?;
+    let pool = Pool::builder().build(manager).await?;
+
+    let _ = REDIS_POOL
+        .write()
+        .map(|mut x| *x = Some(pool))
+        .map_err(|e| return anyhow!("Can't write redis pool: {}", e));
+    CLIENT_ID
+        .write()
+        .map(|mut x| *x = client_id.clone())
+        .map_err(|e| anyhow!("Can't write client id: {}", e))
+}
+
+#[allow(dead_code)]
+async fn log_to_redis(log: &String) -> anyhow::Result<()> {
+    let pool = REDIS_POOL
+        .read()
+        .map_err(|e| anyhow!("Can't read redis pool: {}", e))?;
+    let client_id = CLIENT_ID
+        .read()
+        .map_err(|e| anyhow!("Can't read client id: {}", e))?;
+
+    match pool.clone() {
+        Some(pool) => {
+            let mut conn = pool.get().await?;
+            let _ = cmd("RPUSH")
+                .arg(client_id.clone())
+                .arg(log.clone())
+                .query_async::<_, ()>(&mut *conn)
+                .await
+                .map_err(|e| anyhow!("Can't set log: {}", e));
+
+            Ok(())
+        }
+        None => Err(anyhow!("Redis pool is not initialized")),
+    }
+}
+
+pub fn init_tracing(path: &Path, pkg_name: &str) -> Result<WorkerGuard> {
```

Comment:
Please write a rustdoc for possible library users. I think we can refer to the style of tracing library.
It would be beneficial to mention `_` dropping behavior we discussed in slack.